apps/system/test/unit/app_usage_metrics_test.js

```diff
@@ -195,32 +198,53 @@ suite('AppUsageMetrics:', function() {
         })));
       });
 
-      test('invalid apps', function() {
-        assert.ok(!metrics.shouldTrackApp(null));
-        assert.ok(!metrics.shouldTrackApp(invalidApp));
+      test('don\'t track invalid apps', function() {
+        invalidApps.forEach(function(app) {
+          assert.ok(!metrics.shouldTrackApp(app));
+        });
       });
 
       test('doesn\'t record install/uninstall', function() {
-        var recorded = metrics.recordInstall(invalidApp);
-        assert.equal(getInvalidAppUsage().installs, 0);
-        assert.ok(!recorded);
-
-        recorded = metrics.recordInstall(invalidApp);
-        assert.equal(getInvalidAppUsage().uninstalls, 0);
-        assert.ok(!recorded);
+        invalidApps.forEach(function(app) {
+            var recorded = metrics.recordInstall(app);
+            assert.equal(getUsage(metrics, app).installs, 0);
+            assert.ok(!recorded);
+
+            recorded = metrics.recordInstall(invalidApp);
```

Comment:
I think this is a mistake. Shouldn't it be using app, not invalidApp?

And don't you mean to call recordUninstall() instead of recordInstall() again?  Looks like that error was in the original code, too.